tools/tests/test_late_discovery.py

```
@@ -0,0 +1,165 @@
+
+    source = (
+        Path(__file__).resolve().parents[2]
+        / "custom_components" / "ha_creality_ws" / "number.py"
+    ).read_text()
+    gate = source.split("def _chamber_entities()", 1)[1].split("\n    ents.extend", 1)[0]
+    assert "_cached_has_chamber_control" in gate, (
+        "the capability must be read inside the gate, not captured at setup"
+    )
+    # Live telemetry promotes the capability the same way __init__ does when it
+    # caches it, so a printer that was off at setup still gets the entity.
+    assert "targetBoxTemp" in gate and "maxBoxTemp" in gate
```

Ruff (0.16.1)

[warning] Assertion should be broken down into multiple parts

Break down assertion into multiple parts

(PT018)